### PR TITLE
Use MemoryLoadLibrary

### PR DESCRIPTION
**Pull request by IllusionPerdu**
It's work but code is probably not perfect ;)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,13 +9,14 @@ license = "MIT"
...
 default = ["embed-dll"]
-embed-dll = []
+embed-dll = ["memory-module-sys"]
 
 [dependencies]
 winapi = { version = "0.3.8", features = ["combaseapi", "libloaderapi"] }
 widestring = "0.4.0"
 # com = "0.1"
 com = { git = "https://github.com/microsoft/com-rs.git" }
+memory-module-sys = { git="https://github.com/indygreg/rs-memory-module-sys", branch = "main", optional = true }
```
Comment (sopium) [inline Cargo.toml:19]: Use the crates.io version?

```suggestion
memory-module-sys = { version = "0.3.0", optional = true }
```

File: src/lib.rs [hunk 2]
```diff
@@ -233,6 +238,15 @@ pub struct Stream {
...
 #[derive(Default)]
+#[cfg(feature = "embed-dll")]
+pub struct EnvironmentBuilder<'a> {
+    browser_executable_folder: Option<&'a Path>,
+    user_data_folder: Option<&'a Path>,
+    additional_browser_arguments: Option<&'a str>,
+}
+
+#[derive(Default)]
+#[cfg(not(feature = "embed-dll"))]
```
Comment (sopium) [inline src/lib.rs:249]: You don't need to duplicate the struct definition. The attribute can be applied to the `dll_file_path` field.

File: src/lib.rs [hunk 3]
```diff
@@ -279,6 +293,7 @@ impl<'a> EnvironmentBuilder<'a> {
     ///   It will be simply passed to `LoadLibraryW`.
     ///
     /// Default value: `WebView2Loader.dll`.
+    #[cfg(not(feature = "embed-dll"))]
```
Comment (sopium) [inline src/lib.rs:296]: Can you update the documentation of this function?

The documentation of this whole module (at the start of the source file) also mentions the handling of the DLL file, it need to updated too.

Comment (sopium) [reply src/lib.rs:296]: It could be very confusing that a feature makes a function unavailable.

Maybe we don't need the feature flag at all. Just use the embeded DLL by default, and load an external DLL if this function is called.

Comment (IllusionPerdu) [reply src/lib.rs]: The problem is that if the features are no longer used the executable is bigger for nothing.

Comment (sopium) [reply src/lib.rs]: With LTO, dead code elimination should be able to eliminate the DLL content if it is never used.

Comment (IllusionPerdu) [reply src/lib.rs]: In the last commit i review the logic …
if the "embed-dll" :
 * We use the embedded dll if there is no dll_file_path defined
 * To use a specific dll we must use with_dll_file_path
else :
 * it's like before

File: Cargo.toml [hunk 4]
```diff
@@ -16,6 +16,10 @@ winapi = { version = "0.3.8", features = ["combaseapi", "libloaderapi"] }
...
 # com = "0.1"
 com = { git = "https://github.com/microsoft/com-rs.git" }
+memory-module-sys = { version = "0.3.0" }
 
 [dev-dependencies]
-winit = "0.20.0"
+winit = "0.22.0"
+# futures = "0.3.4"
+# tungstenite = "0.10.1"
+ws = { git = "https://github.com/peddermaster2/ws-rs.git" }
```
Comment (sopium) [inline Cargo.toml:25]: Were you experimenting with some other example but forgot to revert the changes to `dev-dependencies`?

File: src/lib.rs [hunk 5]
```diff
@@ -52,6 +52,8 @@ const DLL: &[u8] =
 #[cfg(all(feature = "embed-dll", target_arch = "aarch64"))]
 const DLL: &[u8] =
     include_bytes!("..\\Microsoft.Web.WebView2.0.9.430\\build\\arm64\\WebView2Loader.dll");
+#[cfg(not(feature = "embed-dll"))]
+const DLL: &[u8] = b"";
```
Comment (sopium) [inline src/lib.rs:56]: If you use `#[cfg(...)]` instead of `if cfg!(...)` in the `build` method, this shouldn't be necessary.

File: Cargo.toml (hunk 4 above)
Comment (IllusionPerdu) [reply Cargo.toml:25]: Yes sorry i have some test for other thing and i don't remove in this pull :s

File: src/lib.rs (hunk 5 above)
Comment (IllusionPerdu) [reply src/lib.rs:56]: thank you for all your comments
